### PR TITLE
Add death timer for level.

### PR DESCRIPTION
**Pull request by dmitriypereverza**
Related to https://github.com/ufocoder/fps/issues/22#issue-2356546593

Add timer component and timerSystem.
Timers can be used in different ways. In this case we add timer for level.

image

### CODE REVIEW

File: src/lib/ecs/systems/WeaponSystem.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ export default class WeaponSystem extends System {
       }
 
       const entityHealth = entityContainer.get(HealthComponent);
+      if (!entityHealth) return;
```
Comment (dmitriypereverza) [inline src/lib/ecs/systems/WeaponSystem.ts:113]: In case we shoot at high speed collision happens on dead enemies.

File: src/global.d.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ interface Level {
         x: number;
         y: number;
     };
+    timer?: number;
```
Comment (ufocoder) [inline src/global.d.ts]: Let's improve the level interface and add an ending scenario field:

```
interface ExitEndingScenario {
  name: 'exit';
  position: {
    x: number;
    y: number;
  }
}

interface EnemyEndingScenario {
  name: 'enemy';
}

interface TimerEndingScenario {
  name: 'timer';
  timer: number;
}

interface Level {
  // ..
  endingScenario: ExitEndingScenario | EnemyEndingScenario | TimerEndingScenario;
  // ..
}
```
Then let's modify all related codebases with this new field

Comment (dmitriypereverza) [reply src/global.d.ts]: done with it

File: src/global.d.ts [hunk 3]
```diff
@@ -14,14 +14,14 @@ interface Item {
 }
 
 
-interface Сharacter {
```
Comment (dmitriypereverza) [inline src/global.d.ts:17]: "C" char has been cyrillic

**Comment by dmitriypereverza** [comment]
@ufocoder done with refactor. I removed timers from ECS and replace with LevelState object in SceneLevel

**Review by ufocoder — APPROVED** [review]

**Pull request merged**